iosVideo/iosVideo/EditVideoViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  EditVideoViewController.swift
+//  iosVideo
+//
+//  Created by Karen Rodriguez on 5/6/20.
+//  Copyright © 2020 Hector Ledesma. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+class EditVideoViewController: UIViewController {
+
+    var player: AVPlayer!
+
+    var videoURL: URL? {
+        didSet {
+            playVideo()
+        }
+    }
+
+    @IBOutlet weak var playerView: VideoPlayerView!
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+        playVideo()
+    }
+
+    private func playVideo() {
+        if isViewLoaded {
+            guard let url = videoURL else { return }
+            player = AVPlayer(url: url)
+            playerView.player = player
+            player.play()
+            NotificationCenter.default.addObserver(forName: .AVPlayerItemDidPlayToEndTime, object: self.player.currentItem, queue: .main) { [weak self] _ in
+                self?.player?.seek(to: CMTime.zero)
+                self?.player?.play()
+            }
+        }
+    }
```

Comment:
Because all of the logic of this method is within that `if` block, we can "flatten" the indentation level of it a bit by combining that `if` logic with the following `guard` statement like so:

```swift
guard isViewLoaded, let url = videoURL
    else { return }
player = AVPlayer(url: url)
// ...
```